### PR TITLE
chore(zh-cn): fix redirects to 'Groups and backreferences'

### PR DESCRIPTION
**Pull request by jasonren0403**
### Related issues and pull requests

Follow up of #18045

### CODE REVIEW

File: files/zh-cn/web/javascript/guide/regular_expressions/assertions/index.md [hunk 1]
```diff
@@ -271,9 +270,9 @@ console.log('This is a First peach in a month.'.match(regex)); // null
 console.log(/\d+(?!\.)/g.exec('3.141')); // [ '141', index: 2, input: '3.141' ]
 ```
 
-### 不同含义的'?!'：断言和范围的组合用法
+### 不同含义的'?!'：断言和字符集的组合用法
```
Comment (yin1999) [inline files/zh-cn/web/javascript/guide/regular_expressions/assertions/index.md:273]: Minor:

```suggestion
### 不同含义的“?!”：在断言和字符集中的组合用法
```

File: files/zh-cn/web/javascript/guide/regular_expressions/assertions/index.md [hunk 2]
```diff
@@ -271,9 +270,9 @@ console.log('This is a First peach in a month.'.match(regex)); // null
...
 
-不同含义的 `?!` 结合使用[断言](/zh-CN/docs/Web/JavaScript/Guide/Regular_expressions/Assertions) `/x(?!y)/` 和[范围](/zh-CN/docs/Web/JavaScript/Guide/Regular_expressions/Groups_and_Ranges) `[^?!]`。
+在像 `/x(?!y)/` 这样的断言和 `[^?!]` 这样的[字符类](/zh-CN/docs/Web/JavaScript/Guide/Regular_expressions/Character_classes)中，`?!` 组合具有不同的含义。
 
 ```plain
```
Comment (yin1999) [inline files/zh-cn/web/javascript/guide/regular_expressions/assertions/index.md:277]: Maybe we also need to format the code block (please also check the next section and the above sections)

````suggestion
```js
````

File: files/zh-cn/web/javascript/reference/regular_expressions/unicode_character_class_escape/index.md [hunk 3]
```diff
@@ -1,5 +1,5 @@
 ---
-title: Unicode property escapes
+title: Unicode 字符类转义：\\p{...}、\\P{...}
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/regular_expressions/unicode_character_class_escape/index.md:2]: We haven't wrapped this with quotation marks

```suggestion
title: Unicode 字符类转义：\p{...}、\P{...}
```

File: files/zh-cn/web/javascript/reference/regular_expressions/unicode_character_class_escape/index.md [hunk 4]
```diff
@@ -13,19 +13,14 @@ slug: Web/JavaScript/Reference/Regular_expressions/Unicode_character_class_escap
 
 > **备注：** 某些 Unicode 属性比[字符类](/zh-CN/docs/Web/JavaScript/Guide/Regular_expressions/Character_classes)(如 `\w` 只匹配拉丁字母 `a` 到 `z`) 包含更多的字符，但后者浏览器兼容性更好（截至 2020 一月）。
 
-## 句法
+## 语法
 
 ```js
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/regular_expressions/unicode_character_class_escape/index.md:18]:
````suggestion
```regex
````

File: files/zh-cn/web/javascript/guide/regular_expressions/assertions/index.md [hunk 5]
```diff
@@ -188,119 +187,120 @@ slug: Web/JavaScript/Guide/Regular_expressions/Assertions
...
 // 使用正则 /^A/ 选择以'A'开头的水果。
 // 这里的 '^' 只有一种含义：匹配输入的开头。
 
-let fruitsStartsWithA = fruits.filter(fruit => /^A/.test(fruit));
+const fruitsStartsWithA = fruits.filter((fruit) => /^A/.test(fruit));
 console.log(fruitsStartsWithA); // [ 'Apple', 'Avocado' ]
 ```
 
-在第二个示例中，^用于在输入的开始处匹配，以及在内部使用时用于创建否定或被补充的字符集[组和范围](/zh-CN/docs/Web/JavaScript/Guide/Regular_expressions/Groups_and_Ranges)。
+在第二个示例中，`^` 用于在输入的开始处匹配，以及在内部使用时用于创建否定或被补充的字符集[组和范围](/zh-CN/docs/Web/JavaScript/Guide/Regular_expressions/Groups_and_Ranges)。
```
Comment (jasonren0403) [inline files/zh-cn/web/javascript/guide/regular_expressions/assertions/index.md:226]:
```suggestion
在第二个示例中，`^` 既用于在输入开头进行匹配，也用于在[字符类](/zh-CN/docs/Web/JavaScript/Guide/Regular_expressions/Character_classes)中使用时创建字符类补集。
```

**Review by yin1999 — APPROVED** [review]
Thank you @jasonren0403

File: files/zh-cn/web/javascript/guide/regular_expressions/assertions/index.md [hunk 6]
```diff
@@ -188,119 +187,120 @@ slug: Web/JavaScript/Guide/Regular_expressions/Assertions
...
-buggyMultiline = buggyMultiline.replace(/^t/gim,'h');
-console.log(1, buggyMultiline); // 修复 'tey'=>'hey'(字符串开始) , 'tangs'=>'hangs'(换行后)
+// 1) 使用 $ 修正字符串结尾处的匹配。
+buggyMultiline = buggyMultiline.replace(/^t/gim, "h");
+console.log(1, buggyMultiline); // 修复 'tey' => 'hey' 和 'tangs' => 'hangs'，而不对 'traa' 做改动。
 
-// 2) 使用 $ 修正字符串结尾处的匹配。
-buggyMultiline = buggyMultiline.replace(/aa$/gim,'ee.');
-console.log(2, buggyMultiline); // 修复 'traa' => 'tree'.
+// 2) Use $ to fix matching at the end of the text.
```
Comment (jasonren0403) [inline files/zh-cn/web/javascript/guide/regular_expressions/assertions/index.md:199]:
```suggestion
// 2) 使用 $ 修正字符串结尾处的匹配。
```

**Pull request merged**